examples/input/keyboard_input_events.rs

```diff
@@ -10,6 +10,18 @@ fn main() {
 /// This system prints out all keyboard events as they come in
 fn print_keyboard_event_system(mut keyboard_input_events: EventReader<KeyboardInput>) {
     for event in keyboard_input_events.iter() {
-        info!("{:?}", event);
+        let scan_code: u32 = event.scan_code;
+        let key_code: Option<KeyCode> = event.key_code;
+        let state: ElementState = event.state;
+        info!("Scan code: {}", scan_code);
+        info!("Key code: {:?}", key_code);
+        match state {
```

Comment:
```suggestion
        info!("Scan code: {}", event.scan_code);
        info!("Key code: {:?}", event.key_code);
        match event.state {
```

Reply:
I chose to declare those variables to make it obvious what type each field is. It's a little more verbose, but I think it adds a bit of clarity.

Reply:
I think I slightly prefer Francois's approach, but it's just a matter of style. I do like how your code lays out the types explicitly though.

Reply:
I also prefer the "inline style" recommended by @mockersf. Thats the style I would encourage developers to use, and I like examples to encourage "best practices" where possible. Minimalism / keeping examples small is also a priority.

Reply:
Done.